h/schemas/pagination.py

```diff
@@ -0,0 +1,16 @@
+from colander import Integer, Range, Schema, SchemaNode
+
+
+class PaginationQueryParamsSchema(Schema):
+    offset = SchemaNode(
+        Integer(),
+        name="page[offset]",
+        validator=Range(min=0),
+        missing=0,
+    )
+    limit = SchemaNode(
+        Integer(),
+        name="page[limit]",
+        validator=Range(min=1, max=500),
+        missing=200,
+    )
```

Comment:
h uses `jsonschema` to validate API requests but `jsonschema` can only validate the JSON bodies of requests, it can't validate other parts of the request like query params, headers or parts of the URL path. As a result, different API views in h have taken different approaches to validating query params:

* The list-groups API view reads query params directly without validation ([here](https://github.com/hypothesis/h/blob/85adcd0b0761765a3651d3d2b469e461ed0fa4f9/h/views/api/groups.py#L24) and [here](https://github.com/hypothesis/h/blob/85adcd0b0761765a3651d3d2b469e461ed0fa4f9/h/views/api/groups.py#L29-L30))
* The get-group API also [reads query params without validation](https://github.com/hypothesis/h/blob/85adcd0b0761765a3651d3d2b469e461ed0fa4f9/h/views/api/groups.py#L101)
* The annotation search API [passes](https://github.com/hypothesis/h/blob/85adcd0b0761765a3651d3d2b469e461ed0fa4f9/h/views/api/annotations.py#L45-L46) `request.params` through a `Colander` schema [`SearchParamsSchema`](https://github.com/hypothesis/h/blob/85adcd0b0761765a3651d3d2b469e461ed0fa4f9/h/schemas/annotation.py#L323-L491) with the help of a util function [`validate_query_params()`](https://github.com/hypothesis/h/blob/85adcd0b0761765a3651d3d2b469e461ed0fa4f9/h/schemas/util.py#L49-L71). This is the only place in the code where either `SearchParamsSchema` or `validate_query_params()` is used.

(This is part of the reason why LMS chose to use `webargs` instead of `jsonschema`: `webargs` can validate and deserialize all parts of a request, and it has Pyramid integration too.)

Here I've gone for the same approach as the annotation search API: defined a Colander schema for the query params and had the body code of the view function use `validate_query_params()` to validate the query params with `PaginationQueryParamsSchema`.

This schema can be re-used by any future views that want to use the same query params.